### ISSUE TITLE
Crashed with Carpet-TIS-Additions (1.19.4)

### ISSUE THREAD

**Issue by NotEvenBelow** [tr7zw/Exordium#129]
mod list:
![294548774-e5479cf4-3afe-4a13-821a-9def30894477](https://github.com/tr7zw/Exordium/assets/104668131/cd9b726b-39fa-477d-a733-9043448e3aba)

log: 
[latest.log](https://github.com/tr7zw/Exordium/files/13847828/latest.log)

**Comment by tr7zw** [comment]
> [23:11:33] [main/ERROR]: Mixin apply for mod carpet-tis-addition failed carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition -> net.minecraft.class_3218: org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException @ModifyExpressionValue expression value modifier method net/minecraft/class_3218::recordBlockState from carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition has an invalid signature. Found unexpected argument type com.llamalad7.mixinextras.sugar.ref.LocalRef at index 1, expected net.minecraft.class_1919. Handler signature: (Lnet/minecraft/class_2680;Lcom/llamalad7/mixinextras/sugar/ref/LocalRef;)Lnet/minecraft/class_2680; Expected signature: (Lnet/minecraft/class_2680;Lnet/minecraft/class_1919;)Lnet/minecraft/class_2680; [INJECT Applicator Phase -> carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition -> Apply Injections ->  -> Inject -> carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition->@ModifyExpressionValue::recordBlockState(Lnet/minecraft/class_2680;Lcom/llamalad7/mixinextras/sugar/ref/LocalRef;)Lnet/minecraft/class_2680;]
org.spongepowered.asm.mixin.injection.throwables.InvalidInjectionException: @ModifyExpressionValue expression value modifier method net/minecraft/class_3218::recordBlockState from carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition has an invalid signature. Found unexpected argument type com.llamalad7.mixinextras.sugar.ref.LocalRef at index 1, expected net.minecraft.class_1919. Handler signature: (Lnet/minecraft/class_2680;Lcom/llamalad7/mixinextras/sugar/ref/LocalRef;)Lnet/minecraft/class_2680; Expected signature: (Lnet/minecraft/class_2680;Lnet/minecraft/class_1919;)Lnet/minecraft/class_2680; [INJECT Applicator Phase -> carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition -> Apply Injections ->  -> Inject -> carpet-tis-addition.mixins.json:logger.microtiming.events.blockevent.ServerWorldMixin from mod carpet-tis-addition->@ModifyExpressionValue::recordBlockState(Lnet/minecraft/class_2680;Lcom/llamalad7/mixinextras/sugar/ref/LocalRef;)Lnet/minecraft/class_2680;]

It's a mixin error from Carpet-TIS-Additions trying to inject into the Server World class, which has nothing to do with Exordium.

**Comment by Fallen-Breath** [comment]
Here the reason why this issue is related to Exordium: exordium shadows MixinExtras without relocates its package. This is not the correct way to use mixin extras, and will cause incompatibility with other mods that [includes](https://github.com/LlamaLad7/MixinExtras?tab=readme-ov-file#setup) MixinExtras

See also: https://github.com/TISUnion/Carpet-TIS-Addition/issues/134#issuecomment-1878986423, https://github.com/TISUnion/Carpet-TIS-Addition/issues/121#issuecomment-1806839438

**Comment by NotEvenBelow** [comment]
mod is also causing crash with other mods

**Comment by NotEvenBelow** [comment]
Exordium also crash when using either MemoryLeakFix or ServerCore

[latest.log](https://github.com/tr7zw/Exordium/files/13848491/latest.log)

**Comment by tr7zw** [comment]
This has been fixed in 1.20.1+, so either uninstall Exordium or update Minecraft.
<details> 
  <summary> I'll be quite bland here: I'm fed up with this mod. (Rant hidden)  </summary>
Exordium was a test! to speed up vanilla Minecraft/lightly modded vanilla gameplay, as it's inherently incompatible with other mods. The mod page says that it's WIP and that it won't work with all mods, and that I won't be fixing that. Yet, some people seem to hold it to the same standards like actually finished mods and keep throwing so many mods that do stuff with the UI at it, followed by then complaining, I kind of don't care anymore at this point. It's also one of the reasons I stopped updating old versions(like <=1.19.4) and stopped trying to get the Forge version to work right. Too much work just for people to complain about stuff that it was not intended to work with/stuff that wasn't finished/got removed(looking at you, #96 ). Not sure if I'll make a 1.20.4 version at all at this point, might as well wait for 1.20.5 where Mojang finally reworked the UI logic to make more sense, and then delist all old versions of the mod so people stop bringing up issues like this one with an old version. Maybe someone else wants to pick up this mod, as I just get too depressed/annoyed even thinking about it, and much rather work on literally anything else.
</details>

Right now the correct fix is to uninstall Exordium and not to use it with mods that don't work with it, especially the old versions(anything <1.2.x). Just install ImmediatelyFast which will get you most of the way there in terms of performance, and doesn't have the same issues.
